### PR TITLE
[tests-only][full-ci] Removed spaces example for public webdav api

### LINKED ISSUE #40030: spaces public WebDAV API missing step
https://drone.owncloud.com/owncloud/ocis/11075/31/6
```
--- FeatureContext has missing steps. Define them with these snippets:

    /**
     * @When the public uploads file :arg1 with content :arg2 using the spaces public WebDAV API
     */
    public function thePublicUploadsFileWithContentUsingTheSpacesPublicWebdavApi($arg1, $arg2)
    {
        throw new PendingException();
    }
```

There is a step written in a feature file somewhere that tries to mention "spaces public WebDAV API".

- find it
- either fix the step ("spaces public WebDAV API" is not really a thing?)
- or make the test code work for that step

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR has removed spaces example for the public WebDAV API version and also refactored `webdav_api_version` to `public-webdav-api-version`

## Related Issue

- Fixes https://github.com/owncloud/core/issues/40030

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -648,7 +648,7 @@ public static function getDavPath(
 		if ($type === "public-files" || $type === "public-files-old") {
 			return "public.php/webdav/";
 		}
-		if ($type === "public-files-new") {
+		if ($type === "public-files-new" || $type === "public-files-spaces") {
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:651]: I think there would be no point in running on different WebDAV versions (new & spaces) if the URLs are same

**Comment by phil-davis** [comment]
The step seems to be only used with "spaces public WebDAV API" in:
https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavLocks/publicLink.feature#L45 (and 46) examples.

Those really are a repeat of the examples at line 43 and 44:
```
    @skipOnOcV10 @personalSpace
    Examples:
      | dav-path | lock-scope | webdav_api_version |
      | spaces   | shared     | old                |
      | spaces   | exclusive  | old                |
      | spaces   | shared     | new                |
      | spaces   | exclusive  | new                |
      | spaces   | shared     | spaces             |
      | spaces   | exclusive  | spaces             |
```
We could just delete those last 2 examples. (And not change any real coe)

"spaces public WebDAV API" is not really a thing.

And adjust the column name in the examples tables. Some scenarios already have `public-webdav-api-version` as the column name. It would be clearer to do that everywhere that a scenario outline has an examples column that is for "public WebDAV API" because "old" and "new" there is a different meaning from "old" and "new" DAV version.

File: tests/acceptance/features/apiWebdavLocks/publicLink.feature [hunk 2]
```diff
@@ -15,35 +15,33 @@ Feature: persistent-locking in case of a public link
...
 
     @skipOnOcV10 @personalSpace
     Examples:
-      | dav-path | lock-scope | webdav_api_version |
-      | spaces   | shared     | old                |
-      | spaces   | exclusive  | old                |
-      | spaces   | shared     | new                |
-      | spaces   | exclusive  | new                |
-      | spaces   | shared     | spaces             |
-      | spaces   | exclusive  | spaces             |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavLocks/publicLink.feature:46]: this was causing the undefined step error, `spaces` was not an option for the step that is using it. :+1:

**Review by kiranparajuli589 — COMMENTED** [review]
overall LGTM :+1:

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**